cmake/modules/libbpf.cmake

```diff
@@ -25,6 +25,8 @@ elseif(NOT USE_BUNDLED_LIBBPF)
         message(FATAL_ERROR "Couldn't find system libbpf")
     endif()
 else()
+    include(zlib)
```

Comment:
they should be already included by `libscap` cmake before arriving here, `include(libbpf)` is done only after the other 2 includes :thinking:

Reply:
Indeed, the issue is `libscap` itself.

A cmake config time, this :point_down: can happen before `include(libelf)` (defined somewhere else in the `libscap` cmake files.

https://github.com/falcosecurity/libs/blob/9d8c0a4babbdd01752d5d86ed6dd23194b339fc0/userspace/libscap/engine/modern_bpf/CMakeLists.txt#L24

The loading order depends on the actual options passed to make. 

In my case, with an older version of libs and the following command:
```
cmake -DUSE_BUNDLED_DEPS=Off -DUSE_BUNDLED_LIBELF=Off -DUSE_BUNDLED_LIBBPF=On -DUSE_BUNDLED_ZLIB=On -DBUILD_FALCO_MODERN_BPF=ON -DBUILD_BPF=ON -DMINIMAL_BUILD=On ..
``` 

I was able to trigger this path:

```
CMake Error at /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3146 (get_property):
  get_property could not find TARGET zlib.  Perhaps it has not yet been
  created.
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3271 (_ep_get_file_deps)
  /usr/share/cmake-3.22/Modules/ExternalProject.cmake:3719 (_ep_add_configure_command)
  build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/cmake/modules/libbpf.cmake:21 (ExternalProject_Add)
  build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/userspace/libscap/engine/modern_bpf/CMakeLists.txt:8 (include)
```

Generally speaking, cmakefiles should be self-contained and not rely on other inclusions that may (or not) happen before. Resolving loading order should be a cmake responsibility, not us :) 


Does it make sense? :thinking: